var/spack/repos/builtin/packages/dbcsr/package.py

```diff
@@ -33,7 +38,8 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('mpi', when='+mpi')
     depends_on('libxsmm@1.11:~header-only', when='smm=libxsmm')
 
-    depends_on('cmake@3.12:', type='build')
+    depends_on('cmake@3.17:', type='build', when='@2.1:')
+    depends_on('cmake@3.10:', type='build', when='@:2.0')
```

Comment:
I think it's easier to reason in terms of increasing requests with increasing versions:
```suggestion
    depends_on('cmake@3.10:', type='build')
```